app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala

```diff
@@ -41,4 +49,14 @@ class HaulierStatusService @Inject()(haulierStatusRepository: HaulierStatusRepos
   def delete(correlationId: CorrelationId): EitherT[Future, DeleteHaulierStatusError, String] =
     EitherT.fromOptionF(haulierStatusRepository.findAndDelete(correlationId), CorrelationIdNotFound)
 
+  def updateStatus()(implicit headerCarrier: HeaderCarrier): Future[Future[HttpResponse]] =
+    haulierStatusRepository.findAllOlderThan(appConfig.notificationCheckSchedulerIntervalSeconds).map {
+      case Nil =>
```

Comment:
let's say TTL gets kicked off at 1pm and scheduler triggers at after 10sec or 20 sec then we get Nil from DB and then it updates status as AVAILABLE  which shows AVAILABLE for a very short span of time in the UI but the service could be UNAVAILABLE time time ?
i think we can ignore as its just an edge case but worth highlighting..

Reply:
Definitely worth highlighting and not the only problem with this approach. We've agreed to go with something very simple initially, but looking to improve it / make it more sophisticated as soon as possible after getting something working end to end.